wgengine/tsdns/tsdns.go

```diff
@@ -401,6 +446,131 @@ func marshalResponse(resp *response) ([]byte, error) {
 	return builder.Finish()
 }
 
+var (
+	rdnsv4Suffix = []byte("in-addr.arpa.")
+	rdnsv6Suffix = []byte("ip6.arpa.")
+)
+
+func rdnsNameToIPv4(name []byte) (ip netaddr.IP, ok bool) {
```

Comment:
add a little comment with an example input?

Maybe easier implementation is just removing the suffix, using netaddr.ParseIP and rearranging the bytes and calling netaddr.IPv4 again?

Couple more allocations than your way, but way less code and this isn't a hot path anyway.